src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -435,6 +436,16 @@ public override void Configure(IApplicationBuilder app)
         }
         else
         {
+            // Force redirect_url to be https, when authentication on BarViz.
+            // Otherwise it deafults to http, since that's how we're talking to Application Gateway
+            app.UseWhen(context => context.Request.Path == AccountController.AccountSignInRoute, app =>
+            {
+                app.Use((context, next) =>
+                {
+                    context.Request.Scheme = "https";
```

Comment:
@dkurepa I understand you have tested this?

I am still wondering how does this affect the code that is feeding the URL to the GH library here:
```csharp
    [HttpGet("/Account/SignIn")]
    [AllowAnonymous]
    public IActionResult SignIn(string returnUrl = null)
    {
        string redirectUrl = Url.Action(nameof(LogInCallback), "Account", new {returnUrl});
        AuthenticationProperties properties =
            SignInManager.ConfigureExternalAuthenticationProperties(Startup.GitHubScheme, redirectUrl);
        return Challenge(properties, Startup.GitHubScheme);
    }
```

This setting influences the `Url.Action` behaviour?

Reply:
I have tested it, yes
No, the `Url.Action` method just returns `Account/LogInCallback?returnUrl=...` the full URL is generated later, down the live by aspnet

Reply:
How I understand this, is that we're "tricking" the server here, by setting the schema as https before it reaches the controller, so the `redirect_uri` parameter is also set to it later

Reply:
https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-7.0#forward-the-scheme-for-linux-and-non-iis-reverse-proxies

Reply:
The problem as I see it is this:

```mermaid
flowchart
    Maestro--1. Goes to GitHub OAuth\n with redirect_url=...-->GitHub
    GitHub--2. User signs in-->GitHub
    GitHub--3. GitHub redirects to redirect_url\nhttp://maestro.dot.net/SignIn..-->AppGateway
    AppGateway--4. Talks to Maestro over HTTP-->Maestro
    Maestro--5. ASP.NET Core changes HTTP to HTTPS inside the pipeline-->Maestro
```

This PR, in my mind, changes 5. (or adds 5.).
But we care about 3.

Reply:
![image](https://github.com/dotnet/arcade-services/assets/7013027/abe5410b-5dce-4bd9-bf55-3cde053a4929)

Reply:
@dkurepa can you confirm that this is not happening? And that we're actually changing the `redirect_url` to have HTTPS?

Reply:
What we're doing with this PR is making it so that the HTTP call doesn't happen because it's dangerous :)